testing/screenshots/make-screenshot.js

```diff
@@ -0,0 +1,242 @@
+      '. Use one of: ' + deviceList.join(',') +
+      '. Default is ' + DEFAULT_DEVICE_ID);
+  phantom.exit(1);
+}
+
+
+var page = require('webpage').create();
+
+// Viewport and User Agent.
+page.viewportSize = !LANDSCAPE_MODE ? DEVICE.viewportSize :
+    {width: DEVICE.viewportSize.height, height: DEVICE.viewportSize.width};
+page.settings.userAgent = DEVICE.userAgent;
+console./*OK*/log('VIEWPORT = ' + page.viewportSize.width + 'x' +
+    page.viewportSize.height);
+console./*OK*/log('USER AGENT = ' + page.settings.userAgent);
+
+
+// Logging.
+var RESOURCE_PREFIX = 'RESOURCE:';
+
+function log(tag, message) {
+  console./*OK*/log(tag, message);
+}
+
+function logFromPage(channel, message) {
+  if (message && message.substring(0, RESOURCE_PREFIX.length) ==
+          RESOURCE_PREFIX) {
+    log('[RESOURCE ' + channel + ']', message.substring(
+        RESOURCE_PREFIX.length));
+  } else {
+    log('[PAGE ' + channel + ']', message);
+  }
+}
+
+page.onResourceRequested = function (request) {
+};
+page.onConsoleMessage = function(message) {
+  logFromPage('INFO', message);
+};
+page.onError = function(message, trace) {
+  logFromPage('ERROR', message);
+};
+
+
+// Rendering.
+function renderAndExit(exitCode) {
+  page.render(FILE);
+  phantom.exit(exitCode);
+}
+
+page.onCallback = function(data) {
+  if (data.event == 'DONE') {
+    renderAndExit(0);
+  }
+};
+
+var self = this;
+page.open(url, function() {
+  log('Page opened: ' + self.page.url + ': ' + status);
+
+  page.evaluate(function() {
+
+    function log(message) {
+      console./*OK*/log('RESOURCE:' + message);
+    }
+
+    function removeFromArray(array, item) {
+      var index = array.indexOf(item);
+      if (index != -1) {
+        array.splice(index, 1);
+      }
+    }
+
+    function prepareResource(resource) {
+      // TODO(dvoytenko): do preparations, e.g. replace `src`, `srcset`, etc.
+    }
+
+    function layoutAll(resourcesManager) {
+      // TODO(dvoytenko): do a repeated calls until no new elements are loaded.
+      window.AMP_RESOURCES_QUEUE = resourcesManager.get();
+      window.AMP_LOG = log;
+      var resources = window.AMP_RESOURCES_QUEUE.slice(0);
+      log('Resources to be loaded: ' + resources.length);
+      var layoutsDone = 0;
+      function completeResource(resource, failure) {
+        layoutsDone++;
+        if (!failure) {
+          log('Resource done: ' + layoutsDone + ': ' + resource.debugid);
+        } else {
+          log('Resource failed: ' + layoutsDone + ': ' + resource.debugid +
+              '; reason: ' + failure);
+        }
+        removeFromArray(window.AMP_RESOURCES_QUEUE, resource);
+        if (layoutsDone >= resources.length) {
+          log('All resources loaded');
+          window.callPhantom({event: 'DONE'});
+        }
+      }
+      resources.forEach(function(resource) {
+        log('Resource started: ' + resource.debugid);
+        prepareResource(resource);
+        resource.forceAll().then(function() {
```

Comment:
I think this is a good test, but how about being able to take a screenshot of just the current viewport after scrolling to some offset?

That would also test the entire incremental loading as well.